python/ray/tests/test_resource_demand_scheduler.py

```diff
@@ -299,6 +380,30 @@ def testScaleUpMinSanity(self):
         autoscaler.update()
         self.waitForNodes(2)
 
+    def testScaleUpMinWorkers(self):
+        config = copy.deepcopy(MULTI_WORKER_CLUSTER)
+        config["min_workers"] = 3
+        config["max_workers"] = 50
+        # Since config["min_workers"] > 2, the remaining worker is started
+        # with the default worker node type.
+        config["available_node_types"]["p2.8xlarge"]["min_workers"] = 2
+        config_path = self.write_config(config)
+        self.provider = MockProvider()
+        runner = MockProcessRunner()
+        autoscaler = StandardAutoscaler(
+            config_path,
+            LoadMetrics(),
+            max_failures=0,
+            process_runner=runner,
+            update_interval_s=0)
+        assert len(self.provider.non_terminated_nodes({})) == 0
+        autoscaler.update()
+        self.waitForNodes(3)
+        assert len(self.provider.mock_nodes) == 3
+        assert self.provider.mock_nodes[0].node_type == "m4.large"
+        assert self.provider.mock_nodes[1].node_type == "p2.8xlarge"
+        assert self.provider.mock_nodes[2].node_type == "p2.8xlarge"
```

Comment:
I think there is another condition you need to handle/test: after the nodes launched are idle > 5 min, the autoscaler will try to kill them. To avoid this, we need to add special logic in the idle node killer. Also, there should be a test for this case (and the test should currently fail, before you add that special logic).

Reply:
Can you look at the updated tests?

Reply:
can you check the updated tests?